runtime/src/iree/hal/drivers/cuda/dynamic_symbols.c

```diff
@@ -47,14 +76,24 @@ iree_status_t iree_hal_cuda_dynamic_symbols_initialize(
   memset(out_syms, 0, sizeof(*out_syms));
   iree_status_t status = iree_dynamic_library_load_from_files(
       IREE_ARRAYSIZE(kCUDALoaderSearchNames), kCUDALoaderSearchNames,
-      IREE_DYNAMIC_LIBRARY_FLAG_NONE, host_allocator,
-      &out_syms->loader_library);
+      IREE_DYNAMIC_LIBRARY_FLAG_NONE, host_allocator, &out_syms->cuda_library);
   if (iree_status_is_not_found(status)) {
     iree_status_ignore(status);
     return iree_make_status(
         IREE_STATUS_UNAVAILABLE,
         "CUDA runtime library not available; ensure installed and on path");
   }
+#if IREE_HAL_DRIVER_CUDA_NCCL
+  status = iree_dynamic_library_load_from_files(
```

Comment:
we don't want to require NCCL - we only need if it's used and can check at that time (is nccl_library NULL, etc) - it is fine for the library to not be present here so long as the user never tries to configure NCCL. failures other than not-found should be reported but not-found is safe to ignore and continue loading.

this is because we want to turn NCCL on as a build option for our release builds installed via pip/etc and allow users to use cuda without requiring them to rebuild IREE

Reply:
This is moved to the channel creation.